xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-default/src/main/java/org/xwiki/refactoring/internal/DefaultModelBridge.java

```diff
@@ -213,7 +216,7 @@ public void createRedirect(DocumentReference oldReference, DocumentReference new
             try {
                 XWikiDocument oldDocument = xcontext.getWiki().getDocument(oldReference, xcontext);
                 int number = oldDocument.createXObject(redirectClassReference, xcontext);
-                String location = this.compactEntityReferenceSerializer.serialize(newReference, oldReference);
+                String location = this.localEntityReferenceSerializer.serialize(newReference);
```

Comment:
Yes using the local serializer is wrong. You should use the default serializer and be done with it, there is no reason to try to "optimize" the reference here.